sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -664,6 +664,10 @@ public void onSuccess(MutateRowResponse produced) {}
   /** Disallow construction of utility class. */
   private BigtableIO() {}
 
+  private static ByteKey makeByteKey(ByteString key) {
+    return ByteKey.copyFrom(key.asReadOnlyByteBuffer());
```

Comment:
Luckily this is really only ever used for metadata operations. And it's not clear to me that `ByteString` won't check `isReadOnly` on the buffer and use it as a view.